### PR TITLE
pugixml ODR fix

### PR DESCRIPTION
**Pull request by triccyx**
Fix for possibile ODR problems on pugixml library.

### CODE REVIEW

**Comment by traversaro** [comment]
The Travis error seems to be: 

> CMake Error: install(EXPORT "blocktestcore" ...) includes target "blocktestcore" which requires target "pugixml" that is not in the export set.

This probably means that `pugixml::pugixml` target needs to be linked with the `blocktestcore` via the `PRIVATE` keyword, to communicate with the build system that the library that link to the `blocktestcore` to dot need to link transitively to `pugixml::pugixml`.

**Review by traversaro — CHANGES_REQUESTED** [review]
I just noticed that the pugixml header is used in the library via `#include "pugixml.hpp"`, meaning that the search path for the header would be first to search in same directory that contains the `.cpp` file, and then in the default include headers paths and the one passed by the `-I` compiler option (opposed to use `#include <pugixml.hpp>`, that will not search in the same directory that contains the `.hpp` file). 

Unfortunately, this means even if `BLOCKTEST_USE_EXTERNAL_PUGIXML` is set to `ON`, the vendored `pugixml.hpp` would be included, being a possible source of inconsistencies. A possible solution is to move the vendored pugixml files in a different directory (tipically used directories are `extern` or something similar) and then add the necessary include directories to `pugixml` in the case `BLOCKTEST_USE_EXTERNAL_PUGIXML` is `OFF`)

**Comment by Nicogene** [comment]
@traversaro did you mean https://github.com/robotology/blocktest/pull/18/commits/61d6ea5b1cd221019f502ccc088e872675a8f3fd ?

**Comment by traversaro** [comment]
> 
> 
> @traversaro did you mean [61d6ea5](https://github.com/robotology/blocktest/commit/61d6ea5b1cd221019f502ccc088e872675a8f3fd) ?

Perfect, even if in that case you also need to tell to the compiler that it needs to search for include files also in `${CMAKE_CURRENT_SOURCE_DIR}/extern/pugixml` otherwise the installation compilation will use the system pugixml includes, and probably will fail in a system without pugixml installed.

**Comment by traversaro** [comment]
> 
> 
> > @traversaro did you mean [61d6ea5](https://github.com/robotology/blocktest/commit/61d6ea5b1cd221019f502ccc088e872675a8f3fd) ?
> 
> Perfect, even if in that case you also need to tell to the compiler that it needs to search for include files also in `${CMAKE_CURRENT_SOURCE_DIR}/extern/pugixml` otherwise the installation compilation will use the system pugixml includes, and probably will fail in a system without pugixml installed.

TL;DR: I think that if in Travis you do not install the `libpugixml-dev` package in https://github.com/robotology/blocktest/blob/master/.ci/install_debian.sh#L13, the build will fail.
